File: elixir/armstrong-numbers/lib/armstrong_number.ex
defmodule ArmstrongNumber do
  @moduledoc """
  Provides a way to validate whether or not a number is an Armstrong number
  """

  @spec valid?(integer) :: boolean
  def valid?(number) do
    digits = Integer.digits(number)

    sum_of_powers = digits 
    |> Enum.map(fn d -> :math.pow(d, length(digits)) end)
    |> Enum.sum()

    sum_of_powers == number

  end
end

File: kotlin/armstrong-numbers/src/main/kotlin/ArmstrongNumber.kt
import kotlin.math.pow

object ArmstrongNumber {

    fun check(input: Int): Boolean {
        val numberOfDigits = input.toString().length
        val sumOfPowers = input.toString().map { digit -> digit.toString().toDouble().pow(numberOfDigits) }.sum()

        return sumOfPowers.toInt() == input
    }

}
